latch/registry/utils.py

```diff
@@ -0,0 +1,343 @@
+        )
+
+    if "array" in registry_type:
+        if not (type(value) is list):
+            raise ValueError(
+                f"{value} is not a list so it cannot be converted into a list"
+            )
+
+        return [to_python_literal(sub_val, registry_type["array"]) for sub_val in value]
+
+    # python_type = to_python_type(registry_type)
+    try:
+        primitive = registry_type["primitive"]
+    except KeyError:
+        raise ValueError("cannot convert to python - malformed registry type")
+
+    if primitive == "enum":
+        try:
+            members = map(clean, registry_type["members"])
+        except KeyError:
+            raise ValueError(
+                "cannot convert to python - malformed registry enum type without"
+                " members"
+            )
+
+        if clean(value) not in members:
+            raise ValueError(
+                f"unable to convert {value} ({clean(value)}) to any of the enum members"
+                f" {members}"
+            )
+
+        return to_python_type(registry_type)._member_map_.get(clean(value))
+
+    if primitive == "blob":
+        typ = LatchFile
+        if (
+            "metadata" in registry_type
+            and "nodeType" in registry_type["metadata"]
+            and registry_type["metadata"]["nodeType"] == "dir"
+        ):
+            typ = LatchDir
+
+        if not (type(value) is dict and ("ldataNodeId" in value)):
+            raise ValueError("cannot convert non-blob value to blob type")
+
+        path_data = execute(
+            """
+            query nodePath($nodeId: BigInt!) {
+                ldataOwner(argNodeId: $nodeId)
+                ldataGetPath(argNodeId: $nodeId)
+            }
+            """,
+            {"nodeId": value["ldataNodeId"]},
+        )
+
+        url = path_data["ldataGetPath"]
+        owner = path_data["ldataOwner"]
+
+        if url == None:
+            raise ValueError("unable to convert to LatchFile/Dir as no url was found")
+
+        account_match = account_root_regex.match(url)
+        mount_match = mount_regex.match(url)
+
+        path = ""
+
+        if account_match is not None:
+            if owner is None:
+                raise ValueError(
+                    "unable to convert to LatchFile/Dir as owner was not found"
+                )
+
+            path = f"latch://{owner}.account{account_match.groupdict()['path']}"
+        elif mount_match is not None:
+            groups = mount_match.groupdict()
+            path = f"latch://{groups['mount']}.mount{groups['path']}"
+        else:
+            raise ValueError(
+                f"unable to convert to LatchFile/Dir as url was malformed: {url}"
+            )
+
+        return typ(path)
+
+    if primitive == "link":
+        if "sampleId" not in value:
+            raise ValueError("unable to convert to Link as row id was not found")
+        try:
+            table_id = registry_type["experimentId"]
+        except KeyError:
```

Comment:
use `.get` or `in`, exceptions are quite heavy-weight in performance and semantics